packages/deployments/contracts/contracts_forge/messaging/RootManagerPropagateWrapper.t.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {RootManagerPropagateWrapper} from "../../contracts/messaging/RootManagerPropagateWrapper.sol";
+import {RootManager} from "../../contracts/messaging/RootManager.sol";
+
+import "../utils/ConnectorHelper.sol";
+
+contract RootManagerPropagateWrapperTest is ForgeHelper {
+  // ============ Errors ============
+  error ProposedOwnable__onlyOwner_notOwner();
+
+  // ============ Events ============
+  event FundsReceived(uint256 amount, uint256 balance);
+
+  event FundsDeducted(uint256 amount, uint256 balance);
+
+  event RootManagerChanged(address rootManager, address oldRootManager);
+
+  // ============ Storage ============
+  RootManagerPropagateWrapper rootManagerWrapper;
+
+  address payable rootManager = payable(address(2));
+
+  function setUp() public {
+    vm.expectEmit(true, true, true, true);
+    emit RootManagerChanged(rootManager, address(0));
+    rootManagerWrapper = new RootManagerPropagateWrapper(rootManager);
+
+    vm.expectEmit(true, true, true, true);
+    emit FundsReceived(0.1 ether, 0.1 ether);
+
+    address alice = address(1);
+    vm.deal(alice, 1 ether);
+    vm.prank(alice);
+    payable(address(rootManagerWrapper)).transfer(0.1 ether);
+  }
+
+  // ============ RootManagerPropagateWrapper.setRootManager ============
+  function test_RootManagerPropagateWrapper__setRootManager_shouldFailIfNotOwner(address caller) public {
+    address owner = rootManagerWrapper.owner();
+    if (caller == owner) {
+      return;
+    }
+
+    vm.expectRevert(ProposedOwnable__onlyOwner_notOwner.selector);
+    vm.prank(caller);
+    rootManagerWrapper.setRootManager(address(42));
+  }
+
+  function test_RootManagerPropagateWrapper_setRootManager__shouldWork() public {
+    vm.expectEmit(true, true, true, true);
+    emit RootManagerChanged(address(42), rootManager);
+
+    address owner = rootManagerWrapper.owner();
+    vm.prank(owner);
+    rootManagerWrapper.setRootManager(address(42));
+  }
+
+  // ============ RootManagerPropagateWrapper.fallback ============
+  function test_RootManagerPropagateWrapper_propagate__shouldWork() public {
```

Comment:
should be `test_RootManagerPropagateWrapper__propagate_shouldWork` lol

Reply:
Fixed.